spec/models/claim/advocate_claim_spec.rb

```diff
@@ -87,36 +87,36 @@
     end
   end
 
-  context 'eligible fee types' do
+  describe 'eligible fee types' do
     subject(:claim) { build(:unpersisted_claim) }
 
-    before(:all) do
-      @bft1 = create(:basic_fee_type, roles: %w[agfs agfs_scheme_9 agfs_scheme_10], description: 'bft1')
-      @bft2 = create(:basic_fee_type, :lgfs, description: 'bft2')
-      @bft3 = create(:basic_fee_type, description: 'bft3')
-      @bft4 = create(:basic_fee_type, roles: %w[agfs agfs_scheme_9], description: 'bft4')
-      @bft5 = create(:basic_fee_type, roles: %w[agfs agfs_scheme_10], description: 'bft5')
-      @mft1 = create(:misc_fee_type, :agfs_scheme_9)
-      @mft2 = create(:misc_fee_type, :lgfs)
-      @mft3 = create(:misc_fee_type, :agfs_scheme_10)
-      @fft1 = create(:fixed_fee_type)
-      @fft2 = create(:fixed_fee_type, :lgfs)
-    end
+    let(:basic_fee_type_schemes_nine_and_ten) { create(:basic_fee_type, roles: %w[agfs agfs_scheme_9 agfs_scheme_10]) }
+    let(:basic_fee_type_no_scheme) { create(:basic_fee_type) }
+    let(:basic_fee_type_scheme_nine) { create(:basic_fee_type, roles: %w[agfs agfs_scheme_9]) }
+    let(:basic_fee_type_scheme_ten) { create(:basic_fee_type, roles: %w[agfs agfs_scheme_10]) }
 
-    after(:all) do
-      clean_database
+    before do
```

Comment:
I much prefer not to have the `before(:all)` and `after(:all)`. However, I understand that this was done to speed up the tests. Have you checked how much this change has affected how long this file takes?

Reply:
I've made that mistake before so I did check this as I was working on it. This change doesn't affect the time taken to run the tests - the affected tests still run in about ~1 second. But your comment prompted me to check the time for running the whole file after all these changes into account, and it is about 6 or 7 seconds than slower than running them against master (typically 44s vs 37s) 😢 

Having done some digging, it looks like [this](https://github.com/ministryofjustice/Claim-for-Crown-Court-Defence/pull/6246/commits/8a040867e589bab6caa60aae3448e4b49ad4d585) commit is slowing things down.

I suspect it's because it moves a lot of expectations out of single examples into multiple `it` blocks, which means  setups are being run more often. I'll see if I can find a way of speeding things up a bit, but other than introduce more `before(:all)` blocks (or reverting the changes) I'm not sure there's much we can do.

Reply:
The solution, which would not be quick or easy, might be to refactor large parts of the test suite so that they are more targeted. I suspect that a lot of time is spent doing setup that is not required in specific tests.

In the short term, I would prefer tests that are understandable over tests that are quick. If the test suite does get too time consuming, however, we may need to revise this.